### DISCUSSION TITLE
About contact point effects

### DISCUSSION THREAD

**Discussion by Octal450** [JSBSim-Team/jsbsim#1190] [Support requests]
Hi,
I have AeroRP located near the gear X position and everything is well, however CG is quite a bit more forward.
The problem is that when the brakes are applied, while the nose is in the air, the nose really gets slammed down hard.

I'm really not sure where to begin with this, because aside from this, everything is well.

If I move the CG aft near AeroRP, this problem stops. So it makes me think the braking/gear/contact-point effects are being applied at CG... is this right?

Kind Regards,
Josh

**Comment by Warning39** [comment]
@Octal450 just to confirm, are only gear brakes being applied? Or are there also aerodynamic brakes like spoilers and/or speed brakes also being applied at the same time? In terms of isolating which components may be contributing to the large pitch moment you're seeing.

**Comment by Octal450** [comment]
Yes. Only gear brakes. The moment you hit the brakes then nose drops.

Josh

**Comment by Warning39** [comment]
What are the `(x,z)` structural coordinates for your cg for the two locations and the main landing gear? Is this for your A320 model?

**Comment by Octal450** [comment]
No, it is not my A320 model.

Note I'm not positive the z position of empty CG is correct. Maybe it's at fault, but it seems "reasonable" but possibly incorrect.

In this coordinate system, the nose of the airplane is (0,0,0).

Main Gear:
```
		<location unit="M">
			<x>24.3891</x>
			<y>-3.02974</y> <!-- and + for right -->
			<z>-3.07523</z>
		</location>
```

Empty CG (where it acts as expected)
```
	<location name="CG" unit="M">
		<x>24.20826</x> <!-- 34.3% MAC -->
		<y>0</y>
		<z>0.0014</z>
	</location>
```

Loaded CG position X 23.596092m. Z is 0.1295146m. (17.4% MAC)

Engines located on the tail, hence the very aft empty CG.

Kind Regards,
Josh

**Comment by Warning39** [comment]
Okay, so my thinking was that the gear brakes produce a retarding force that is below the cg, which will result in a moment that is nose down. Originally you only made mention of the cg moving backwards, so I assumed the cg's Z coordinate was staying fairly constant, which wouldn't explain why there would be a larger nose down moment between the 2 cg locations.

But with your `(x,z)` figures for the two cg locations I see there is a change in the Z coordinate, with the change resulting in a larger moment arm for the brakes, which would explain why there is a larger nose down moment.

Now whether the scale of this nose down moment is accurate is another thing. Maybe the brake power is too large?

**Comment by Octal450** [comment]
Hi,
Sorry - I neglected to mention that when I was testing with empty Cg, it removed the offset of Z that came from altering the weight distribution from default to empty. That is why I only wrote backwards. But you are correct, the Z changed.

OK, let me look into this, this is a good idea and to be honest... something I should have caught. :D

Brake power was calibrated based on performance charts, so it should be right. It happens even with mild autobrakes which do not fully apply it.

Kind Regards,
Josh

**Comment by Warning39** [comment]
For interest here is a patent regarding an autobrake system for aircraft, [Aircraft automatic braking system](https://patentimages.storage.googleapis.com/80/5c/24/7c878146e720d1/US4646242.pdf), which mentions:

> Heretofore, utilization of the autobrake system on aircraft has tended to derotate the aircraft following touchdown at a faster rate than would occur in a landing in which the pilot utilized manual brakes. This characteristic has been considered undesirable by some pilots since they feel that excessive elevator control is required to prevent a hard nose-gear touchdown. This problem has two roots:
>
> (a) Brake application prior to nose-gear touchdown. Application of main-gear brakes will always impose a pitch-down moment on the aircraft. Present autobrake systems will apply brake pressure immediately after main-gear touchdown whereas a pilot will generally not apply manual brakes until after the nose gear is on the ground. Application of even low-to-moderate levels of braking will cause a rapid derotation of the aircraft if the nose gear is not yet on the ground and is unable to react the resultant pitch-down moment on the aircraft. The pitch-down moment prior to nose-gear touchdown can be reacted only by nose-up elevator control.

**Comment by Octal450** [comment]
Thanks for the comment. Yes the aircraft has a 3 second pause on MIN/MED and 1 second on MAX.

I should note that in my other aircraft, the autobrake effect can be overridden by holding elevator back. In this one it can't. I know what you're thinking, maybe Cmde is too weak - it's not, if I increase it's coefficient then the elevator will be way too oversensitive in flight.

Going to need to dig into this a bit more. I also noticed this aircraft has more trouble rotating with the correct stabilizer setting but once airborne, flies in trim. Seems related.

Kind Regards,
Josh

**Comment by Warning39** [comment]
> I know what you're thinking, maybe Cmde is too weak

Nope, my initial guess would be that your brake moment is too large😉 

Remember, the moment being a combination of the brake force applied and the moment arm.  So for example the brake force may be accurate, but if the moment arm is greater than it should be, then you would still have an issue. And then there is the additional variable in terms of your autobrake implementation.

Okay, so in terms of your autobrake logic, I see your mention of different delay periods from WoW. But what do the options, e.g. MAX mean. Does that mean you apply maximum brake force, or does it mean you target some defined maximum deceleration? 

If you take a look at the Boeing patent link above:

> adjusting the deceleration level of said aircraft automatic braking system to said preselected **commanded deceleration level** 

> particular problem is further aggravated by use of an accelerometer for feedback control of deceleration by the autobrake system. 

You'll notice that the pilot selected autobrake option, i.e. MIN, MED or MAX is mapped to a deceleration value, e.g. say -0.1m/s^2 versus -0.2m/s^2 versus -0.3m/s^2. As opposed to be simply mapped to a commanded brake command, e.g. 30%, 60% and 100%.

The difference being that a deceleration command in m/s^2 means that the autobrake system will effectively take into account the current mass of the aircraft and therefore also the current pitch moment of inertia.

So for example, assuming, and I don't know for sure if it is the case in your implementation, but MAX was mapped to 100% brake command, then on touchdown 1 second after WoW a lightly loaded aircraft would experience a much larger nose down pitching moment from the brakes compared to a heavily loaded aircraft.

**Comment by Warning39** [comment]
Some useful figures in this forum with regards to autobrake - https://www.airliners.net/forum/viewtopic.php?t=1475865

> On the B777 the aircraft aims for a specific deceleration, depending on the autobrake selector position.
Autobrake 1 is 4.0ft/sec² with a maximum brake pressure of 1385psi.
2 is 5.0 ft/sec² at max 1600psi
3 is 6.0ft/sec² at max 1850psi
4 is 7.5ft/sec² at max 2150psi
MAX AUTO is 11.0 at max 3100psi
RTO is all you can get at a maximum command pressure of 3100psi.
All settings except RTO have a 0.1sec delay after the ground signal is received.

> The MD11 has 5 settings, OFF, T.O., MIN, MED and MAX.
MIN aims for 6.5ft/sec²
MED aims for 9.0ft/sec²
MAX and T.O. gives you all it can get at 3000psi

> On the A320 the autobrake system kicks in as soon as the ground spoilers deploy. In LOW mode the brake pressure then gradually ramps up over the span of 3sec until it reaches 1.7m/sec² (5.6ft/sec²). MED mode aims for a deceleration of 2m/sec² (6.6ft/sec²) without delay when the ground spoilers deploy which is then increased to 3m/sec² (9.8ft/sec²) when the aircraft pitch is less than 1° or after 4 seconds.
MAX mode aims for 10m/sec² (32.8ft/sec²). This is more than the maximum possible deceleration of the aircraft so the system applies full pressure.

**Comment by Octal450** [comment]
Hi,
Yes already implemented.

Kind Regards,
Josh

**Comment by Warning39** [comment]
So after implementing an acceleration control, with say a MAX near 10ft/sec^2 with a ramp what happens now in terms of the nose down pitching moment?

**Comment by Octal450** [comment]
Sorry, I did not make it clear. Autobrake is a long standing feature in my planes. -7.5 ft/s^2 as per manual for MED setting tested here.

Kind Regards,
Josh

**Comment by Warning39** [comment]
Okay, so if you've already implemented the autobrake functionality using a control system to achieve a specific deceleration rate then your initial query regarding the excessive nose pitch down is still an issue?

I presume your mention of -7.5 m/s^2 for MED setting is a typo and should be ft/s^2, based on the examples listed above for 3 different airlines?

**Comment by Octal450** [comment]
Yes, this control system implemented years ago. This issue occur.

Sorry yes. Been travelling home for Thanksgiving so I'm on mobile. Ft/sec^2.

Kind Regards,
Josh

**Comment by Warning39** [comment]
So based on the Boeing patent I see two potential issues. Can you post a link to your control system?

How are you measuring the deceleration? Are you potentially running into the pitch attitude issue with the gravity vector interfering with your deceleration measurement?

You mentioned a fixed hold-off period before applying the autobrake, i.e. either 1s or 3s depending the option selected. But after this hold-off period do you immediately target the deceleration figure, e.g. -7.5ft/s^2 for your MED case? Or do you have a ramp up to that target deceleration figure over a couple of seconds?

**Comment by Octal450** [comment]
Hi,
Sorry but the autobrake is not the issue here. Same issue occurs with manual brakes (not full, partial pressure). I'm just using the A/B to make testing easier and more consistent.

Deceleration is computed via aircraft groundspeed. The control system designed to slowly increase brake pressure, not rapidly. Is using a feedback control system that slowly applies braking pressure proportionally, and with an integrator for closing steady state error between -7.5 and current deceleration. So it does not simply bang on a hard brake amount.

In my other aircraft, the exact same autobrake control system has no problems like this.

Kind Regards,
Josh

**Comment by Warning39** [comment]
Well, there are only a handful of things contributing to a large nose pitch down moment on application of the brakes. The brake force, the moment arm between the wheels and the cg and the pitch moment of inertia.

So let's see plots of those to see what the values are and how they compare to the other aircraft you mention not having this issue.

> The control system designed to slowly increase brake pressure, not rapidly.

You haven't provided a link to the implementation, so I'm not 100% certain of what you're doing, but based on some of your other comments above it sounds like a PI controller?

But unless you're also controlling the set-point (deceleration target), then surely with a standard PI controller on touchdown the initial error is going to -7.5ft/s^2, multiplied by your proportional gain, so the very initial output in terms of brake pressure is going to be a large spike unless you have a pretty small proportional gain.

But if you provide some plots as I suggested then we can see what is happening and potentially rule some things in or out that are contributing to the large pitching moment.

**Comment by Octal450** [comment]
The delay in brake application is caused by the brake hysteresis itself, so I didn't feel the need to implement one control system side.

But regardless, here is the graphs from FlightGear Phi.

**PLEASE NOTE: By mistake I graphed -kts property instead of fps2. It doesn't matter since there are no numbers shown here, but I am using fps2 in the control system.**

Current: 
The decel command was -7.5 immediately when active-final reaches 1 (the 3-second delay).

![image](https://github.com/user-attachments/assets/2d3ca42a-4fb3-459a-954f-a621cb37fa52)

Now I've implemented hysteresis in decel-cmd. I included this here. I had to keep the initial slope a little bit steep otherwise the existing deceleration causes the ABS command to be delayed too long. This already adds nearly a second of delay to the ABS brake command.
![image](https://github.com/user-attachments/assets/0f24c1d7-5678-47eb-8893-c1c4e5b7a26f)

But regardless, to achieve the required deceleration rate, there is still a large output to the brakes and as a result, and the same thing ultimately happens.

Now I could probably do bunch of work on trying to optimize this but the point I'm making is that this happens with manual braking too! Hence I don't want to have to change the ABS system in tons of ways when this same control system works on multiple other airplanes at various CG's without this issue.

Kind Regards,
Josh

**Comment by Warning39** [comment]
> this same control system works on multiple other airplanes at various CG's without this issue.

JSBSim applies the same physics, it has no knowledge regarding the other aircraft. So what are the differences between this aircraft and the others?

The basic physics in terms of generating a nose down pitching moment from the brakes depends on:

- Brake force
- Moment arm for the brake force
- Pitch moment of inertia

So how do those 3 differ between this aircraft and the other aircraft?

**Comment by Octal450** [comment]
Yeah, this is why I'm stuck.

I'm going to test more CG's. 

Can you confirm that I am correct in saying the location of AeroRP is not relevant here?

Kind Regards,
Josh

**Comment by Warning39** [comment]
The difference between the AeroRP and the current cg is only used with the aerodynamic forces. AeroRP isn't used when calculating the forces and moments of the landing gears.

**Comment by Octal450** [comment]
I've done some rechecking and the CG Z position was too high. I've recomputed it and it seems to have helped. I'm monitoring.

Kind Regards,
Josh
